### PR TITLE
Object id observer

### PR DESCRIPTION
**Pull request by codaishin**

### CODE REVIEW

File: src/plugins/common/src/components/object_id.rs [hunk 1]
```diff
@@ -13,24 +13,28 @@ use uuid::Uuid;
...
-			id.entity = Some(entity);
-		}
+	pub(crate) fn with(mut self, entity: Entity) -> Self {
+		self.entity = Some(entity);
+		self
+	}
+
+	pub(crate) fn entity_is_set(&self) -> bool {
+		self.entity.is_some()
 	}
```
[the remark this answers is not in the document]
Comment (codaishin) [reply src/plugins/common/src/components/object_id.rs:38]: I don't think that is necessary. The method is fairly self explanatory. And documenting who uses this method can quickly become a lie, when this docstring is not updated for new uses in future code changes.

File: src/plugins/common/src/observers/update_object_id.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	#[test]
+	fn update_entity_when_reinserted() {
+		let mut app = setup();
+		let object_id = ObjectId::default();
+
+		let mut entity = app.world_mut().spawn(object_id);
+		entity.insert(object_id);
+
+		assert_eq!(Some(&object_id.with(entity.id())), entity.get::<ObjectId>());
+	}
```
[the remark this answers is not in the document]
Comment (codaishin) [reply src/plugins/common/src/observers/update_object_id.rs:55]: The early return was only added to prevent infinite recursion. The observers reacts to insertion and itself inserts something. So there is no need to test this separately, because if the early return is missing, all other tests would cause a stack overflow panic.
